### PR TITLE
chore: migrate from tap to node:test and c8

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

This PR migrates from `tap` to `node:test` and `c8`

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,11 +39,11 @@
     "fastify": "^5.0.0-alpha.3",
     "snazzy": "^9.0.0",
     "standard": "^17.1.0",
-    "tap": "^18.7.0",
     "tsd": "^0.31.0"
   },
   "dependencies": {
     "abstract-cache": "^1.0.1",
+    "c8": "^10.1.2",
```
Comment (gurgunday) [inline package.json:46]: Can you move it to devDeps

Comment (dancastillo) [reply package.json]: Good catch should be fixed now

**Comment by Uzlopak** [comment]
@dancastillo 

Can you please have a look at your previous migrations and check if c8 is using the `--100` flag everywhere?

**Review by gurgunday — APPROVED** [review]
lgtm

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**